### PR TITLE
[serve] Have serve agent return 501 if dependencies are missing

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Minor change, if we fail to import Serve because dependencies are missing, return a [501 = Not Implemented](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501), rather than just returning 500.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by GeneDer — APPROVED** [review]
:shipit:

File: dashboard/modules/serve/serve_agent.py [hunk 1]
```diff
@@ -55,7 +55,16 @@ async def get_version(self, req: Request) -> Response:
     @routes.get("/api/serve/deployments/")
     @optional_utils.init_ray_and_catch_exceptions()
     async def get_all_deployments(self, req: Request) -> Response:
-        from ray.serve.schema import ServeApplicationSchema
+        try:
+            from ray.serve.schema import ServeApplicationSchema
+        except ImportError:
```
Comment (edoakes) [inline dashboard/modules/serve/serve_agent.py:60]: Rather than copy-pasting the same logic across the handlers, can you create a shared decorator?

Comment (edoakes) [reply dashboard/modules/serve/serve_agent.py:60]: or ideally we could have some kind of shared logic that all routes go through (like a middleware), idk if this HTTP framework can support such a thing

**Review by edoakes — APPROVED** [review]
Thanks :)

**Comment by zcin** [comment]
Hi @zhe-thoughts could we merge this one too? It's a small change, and once it's merged we can more gracefully handle missing serve dependencies from the product side like https://github.com/anyscale/product/issues/22861

**Review by sihanwang41 — APPROVED** [review]

**Review by shrekris-anyscale — APPROVED** [review]

**Review by zhe-thoughts — APPROVED** [review]
OK to merge

**Pull request merged**